src/main/scala/com/gu/configraun/aws/AwsInstanceTags.scala

```diff
@@ -0,0 +1,42 @@
+package com.gu.configraun.aws
+
+import com.amazonaws.regions.{Region, Regions}
+import com.amazonaws.services.ec2.{AmazonEC2, AmazonEC2Client, AmazonEC2ClientBuilder}
+import com.amazonaws.services.ec2.model.{DescribeTagsRequest, Filter}
+import com.amazonaws.util.EC2MetadataUtils
+import com.gu.configraun.Errors.{ConfigException, ConfigraunError}
+
+import scala.collection.JavaConverters._
+
+object AwsInstanceTags {
+
+  lazy val instanceId = EC2MetadataUtils.getInstanceId match {
+    case s:String => Right(s)
+    case _ => Left(ConfigException("Unable to find instance id", null))
+  }
+
+  lazy val region = Regions.getCurrentRegion match {
+    case r:Region => Right(r)
+    case _ => Left(ConfigException("Unable to find region", null))
+  }
+  lazy val ec2Client: Either[ConfigraunError, AmazonEC2] = Right(AmazonEC2ClientBuilder.defaultClient())
```

Comment:
Nitpick: you can omit the empty brackets on `.defaultClient()`. Unless it's exhibiting a side effect.